### PR TITLE
Move pending orders to cancel

### PR DESCRIPTION
**Pull request by enishant**
Added cron to check pending orders with provided timeout and cancel them to update inventory stock.

### CODE REVIEW

File: Model/EnablePendingOrdersCron.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                    setTimeout(function(){
+                                        $('.rzp-pending-order-cron-to-clipboard').text('Copy Cron');
+                                    },5000);
+						        });
+						    });
+						});
+						//]]>
+						</script>
+						";
+        $element->setComment("Setup cronjob at server for moving pending orders to Cancel status after timeout. <br><br>*Please execute following command within Magento root directory to setup cronjob* <br><span style='width:300px;font-weight: bold;' class='rzp-pending-order-cron-job' >php bin/magento setup:cron:run --group=\"razorpay\"</span><br/>" . $copyButton);
```
Comment (ramth05) [inline Model/EnablePendingOrdersCron.php:39]: @enishant , if magento cron already running and setup. It should auto setup. Not sure if this is required.

File: Cron/CancelPendingOrders.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        $this->config                          = $config;
+        $this->logger                          = $logger;
+        $this->isCancelPendingOrderCronEnabled = $this->config->isCancelPendingOrderCronEnabled();
+        $this->pendingOrderTimeout             = ($this->config->getPendingOrderTimeout() > 0) ? $this->config->getPendingOrderTimeout() : 30;
+    }
+
+    public function execute()
+    {
+        // Execute only if Cancel Pending Order Cron is Enabled
+        if($this->isCancelPendingOrderCronEnabled === true && $this->pendingOrderTimeout > 0)
```
Comment (ramth05) [inline Cron/CancelPendingOrders.php]: space required after `if`  and multiple conditions should go to new line.

File: Cron/CancelPendingOrders.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            $this->logger->info("Cronjob: Cancel Pending Order Cron started.");
+            $dateTimeCheck = date('Y-m-d H:i:s',strtotime('-' . $this->pendingOrderTimeout . ' minutes'));
```
Comment (ramth05) [inline Cron/CancelPendingOrders.php]: space after comma

File: Cron/CancelPendingOrders.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            $sortOrder = $this->sortOrderBuilder->setField('entity_id')->setDirection('DESC')->create();
+            $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter(
+                'created_at',
+                $dateTimeCheck,
+                'lt'
+            )->addFilter(
+               'status',
+               'pending',
```
Comment (ramth05) [inline Cron/CancelPendingOrders.php]: @enishant , use constant flag for status `pending`

File: Cron/CancelPendingOrders.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        // Execute only if Cancel Pending Order Cron is Enabled
+        if($this->isCancelPendingOrderCronEnabled === true && $this->pendingOrderTimeout > 0)
+        {
+            $this->logger->info("Cronjob: Cancel Pending Order Cron started.");
+            $dateTimeCheck = date('Y-m-d H:i:s',strtotime('-' . $this->pendingOrderTimeout . ' minutes'));
+            $sortOrder = $this->sortOrderBuilder->setField('entity_id')->setDirection('DESC')->create();
+            $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter(
+                'created_at',
```
Comment (ramth05) [inline Cron/CancelPendingOrders.php]: @enishant , it's good if we can use `updated_at` check.

File: Cron/CancelPendingOrders.php (hunk 3 above)
Comment (ramth05) [reply Cron/CancelPendingOrders.php]: Is it going to works fine in case of different time zones.

File: Cron/CancelPendingOrders.php (hunk 4 above)
Comment (enishant) [reply Cron/CancelPendingOrders.php]: @ramth05 Updated constant flag for status pending

File: Cron/CancelPendingOrders.php (hunk 2 above)
Comment (enishant) [reply Cron/CancelPendingOrders.php]: @ramth05 Updated space after if and multiple conditions to new line.

File: Cron/CancelPendingOrders.php (hunk 3 above)
Comment (enishant) [reply Cron/CancelPendingOrders.php]: @ramth05 Updated space after comma

Comment (enishant) [reply Cron/CancelPendingOrders.php]: @ramth05 Magento system's default timezone is UTC, date function creates date with system's timezone. For creating an order and executing cron timezone must be same, so any timezone will work here. Updated timezone to Asia/Kolkata for testing changes.

File: Cron/CancelPendingOrders.php (hunk 5 above)
Comment (enishant) [reply Cron/CancelPendingOrders.php]: Thanks for suggestion, Updated date check with `updated_at`

File: Model/EnablePendingOrdersCron.php (hunk 1 above)
Comment (enishant) [reply Model/EnablePendingOrdersCron.php:39]: @ramth05 Showing instructions to setup cron will be helpful if cron is not already running on server.

File: Cron/CancelPendingOrders.php (hunk 3 above)
Comment (ramth05) [reply Cron/CancelPendingOrders.php]: @enishant , please verify this field must be set through PHP script not directly in mysql.

**Review by ramth05 — APPROVED** [review]
LGTM, Please test throughly before merging it.

**Comment by gauravgarg-c** [comment]
Verified by Abdul and Nishant. Merging it.

**Pull request merged**